src/routes/jira/security/workspaces/containers/jira-security-workspaces-containers-post.ts

```diff
@@ -0,0 +1,90 @@
+import { Request, Response } from "express";
+import { Errors } from "config/errors";
+import { reverseCalculatePrefix, transformRepositoryId } from "~/src/transforms/transform-repository-id";
+import { GITHUB_CLOUD_BASEURL } from "~/src/github/client/github-client-constants";
+import { RepoSyncState } from "models/reposyncstate";
+
+interface RepoUrlAndRepoId {
+	repoUrl: string,
+	id: string
+}
+
+export interface SecurityContainer {
+	id: string,
+	name: string,
+	url: string,
+	avatarUrl: string,
+	lastUpdatedDate: Date
+}
+
+export const DEFAULT_AVATAR = "https://github.githubassets.com/images/modules/logos_page/GitHub-Mark.png;";
+
+export const splitServerId = (input: string): [string, string] => {
+	const parts: string[] = input.split("-");
+	return [parts[0], parts[1]];
+};
+
+export const getRepoUrlAndRepoId = (id: string): RepoUrlAndRepoId => {
+	if (/-/.test(id)) {
+		const [hashedRepoUrl, repoId] = splitServerId(id);
+		const repoDomain = reverseCalculatePrefix(hashedRepoUrl);
+		return { repoUrl: repoDomain, id: repoId };
+	} else {
+		return { repoUrl: GITHUB_CLOUD_BASEURL, id };
+	}
+};
+
+const getRepos = async (repoIds: string[]): Promise<RepoSyncState[] | []> => {
+	const results = await Promise.all(
+		Array.from(new Set(repoIds)).map(async (id) => {
+			// Account for server repoIds which will be passed in a format similar to "XXXXXXX-XXXX"
+			const { repoUrl, id: repoId } = getRepoUrlAndRepoId(id);
+			return await RepoSyncState.findOneForRepoUrlAndRepoId(repoUrl, repoId);
+		})
+	);
+
+	// Loose check so we filter null and undefined values
+	return results.filter((result) => result != null) as RepoSyncState[];
+};
+
+export const transformRepositories = async (
+	repos: RepoSyncState[]
+): Promise<SecurityContainer[]> => {
+	const transformedSubscriptions = repos.map((repo) => {
+		const { repoId, repoName, repoUrl, updatedAt } = repo;
+		const baseUrl = new URL(repoUrl).origin;
+		return {
+			id: transformRepositoryId(repoId, baseUrl),
+			name: repoName,
```

Comment:
So looking at the PR, my question is:
1. Is the `repoName`, `avatarUrl` ( and `lastUpdatedDate` ) the only info needed by the security module? Which is the whole point of this PR? 
2. Since security module already have `repoId` and `repoUrl`, if `1` is true, and assuming we can modify our app to push those info into DataDepot at the first places, wouldn't that means we don't need this PR at all? As all those information is already available?

I mean, this doesn't seem heaps of (critical) information are missing. Can't we just save the app from calling this endpoint?

Reply:
@gxueatlassian `fetchContainers` and `searchContainers` (to be implemented) both only need to return `id, name, url, avatarUrl, lastUpdatedDate` https://developer.atlassian.com/cloud/jira/software/modules/security-information/

> Can't we just save the app from calling this endpoint?
In an ideal situation, we would like to get to this point eventually. We are looking into how we can store Security Container information in DD but this is not planned for this quarter and needs more investigation.

We already have existing patterns in place on both the Security in Jira page and Toolchain page which rely on calling these actions specified by the security module. This pattern is already being used by 5 different security providers and would like to avoid creating specific rules just for GitHub.

Reply:
Okay got cha, I am good with this now. 👍